components/feature/session/src/test/java/mozilla/components/feature/session/FullScreenFeatureTest.kt

```diff
@@ -4,97 +4,386 @@
 
 package mozilla.components.feature.session
 
-import mozilla.components.browser.session.Session
-import mozilla.components.browser.session.SessionManager
-import mozilla.components.concept.engine.EngineSession
+import android.view.WindowManager
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.action.TabListAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.libstate.ext.waitUntilIdle
 import mozilla.components.support.test.mock
-import mozilla.components.support.test.whenever
+import org.junit.After
+import org.junit.Assert.assertEquals
 import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotEquals
+import org.junit.Assert.assertNull
 import org.junit.Assert.assertTrue
+import org.junit.Before
 import org.junit.Test
-import org.mockito.ArgumentMatchers.anyString
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito.anyString
+import org.mockito.Mockito.doReturn
 import org.mockito.Mockito.never
-import org.mockito.Mockito.spy
 import org.mockito.Mockito.verify
 
 class FullScreenFeatureTest {
+    private val testDispatcher = TestCoroutineDispatcher()
 
-    private val sessionManager: SessionManager = mock()
-    private val selectedSession: Session = mock()
-    private val useCases = spy(SessionUseCases(sessionManager))
+    @Before
+    @ExperimentalCoroutinesApi
+    fun setUp() {
+        Dispatchers.setMain(testDispatcher)
```

Comment:
Can `MainCoroutineRule` be used instead here?

Reply:
We should consider refactoring `MainCoroutineRule` to use a `TestCoroutineDispatcher` by default. So that it is easier to get access to methods like `advanceUntilIdle()`.

Reply:
I'll file another issue for that. We can still use the rule here as you can pass the test dispatcher into the rule's constructor.